File: Exemplos/App_IntentDigitalHUB_Ionic/Elgin_IntentDigitalHUB_Ionic-font/src/intentDigitalHubService/BALANCA/Commands/LerPeso.ts
import {BalancaCommand} from '../BalancaCommand';

export class LerPeso extends BalancaCommand {
  readonly qtdLeituras: number;

  constructor(qtdLeituras: number) {
    super('LerPeso');
    this.qtdLeituras = qtdLeituras;
  }

  functionParameters(): object {
    return {
      qtdLeituras: this.qtdLeituras,
    };
  }
}

File: Exemplos M10/App_IntentDigitalHUB_XamarinForms/Elgin_IntentDigitalHUB_XamarinForms-font/E1_Digital_Hub_Xamarin_Forms_M8/IntentServices/Balanca/Commands/LerPeso.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Xamarin_Forms_Intent_Digital_Hub.IntentServices.Balanca.Commands
{
    class LerPeso : BalanceCommand
    {
        readonly private int qtdLeituras;

        public LerPeso(int qtdLeituras) : base("LerPeso")
        {
            this.qtdLeituras = qtdLeituras;
        }

        protected override string FunctionParameters()
        {
            return "\"qtdLeituras\"" + ":" + this.qtdLeituras;
        }
    }
}
